lib/client.js

```diff
@@ -93,6 +138,8 @@ function MqttClient (streamBuilder, options) {
 
   this.options.clientId = (typeof this.options.clientId === 'string') ? this.options.clientId : defaultId()
 
+  this.options.customHandleAcks = this.options.protocolVersion === 5 ? this.options.customHandleAcks : false
```

Comment:
it seems we are manipulating `this.options`  a lot in this function. Would you mind extracting it into a variable `var options = this.options`?

Reply:
fixed